### ISSUE TITLE
Is there a way to hit the local instance programmatically via API or npm package?

### ISSUE THREAD

**Issue by contractorwolf** [nomic-ai/gpt4all-chat#53]
I would love to run node services against this local instance if possible.  Has anyone tried to do that yet?

**Comment by AndriyMulyar** [comment]
The software stack around gpt4all models is growing quickly. Please subscribe to the announcements channel in the discord for the latest updates.
https://discord.gg/ax5T8eyC6a

Describe in more detail below what you are trying to do and it will be taken into account for prioritization!

**Comment by contractorwolf** [comment]
Yeah @AndriyMulyar much like chatGPT also has access to an API for GPT3.5 and 4 I would like to be able to ask the engine questions using JavaScript instead of using the electron(?) interface. I had hoped it was going to be running a normal web server so I could peek it the network traffic and figure out a way to hack a programmatic interface into it, but I don't know how to peek at an electron app. :/

**Comment by manyoso** [comment]
The GUI is Qt based not electron. There will be plugins including javascript, python and even a web REST interface yes in the future.

**Comment by vRobM** [comment]
Take a look at https://github.com/keldenl/gpt-llama.cpp which provides the API interface
